### ISSUE TITLE
Typedef Function generated as null

### ISSUE THREAD

**Issue by skybur** [Milad-Akarie/auto_route_library#35]
I have this widget and I'm using `ValueChanged` from flutter
```
class AddressScreen extends StatefulWidget {
  final ValueChanged<UserAddress> onAddressSelected;
  final bool showCloseAsCross;

  const AddressScreen(
      {Key key, this.showCloseAsCross = false, this.onAddressSelected})
      : super(key: key);
  @override
  AddressScreenState createState() => AddressScreenState();
}
```
and this is the generated argument
```

//**************************************************************************
// Arguments holder classes
//***************************************************************************

//AddressScreen arguments holder class
class AddressScreenArguments{
final Key key;
final bool showCloseAsCross;
final null onAddressSelected;
AddressScreenArguments({
this.key,this.showCloseAsCross = false,this.onAddressSelected});
}

```

I'm using `auto_router` version 0.3.0

**Comment by Milad-Akarie** [comment]
@skybur I'll look into it. Are you sure it's a good idea to pass a callback func to a route though?

**Comment by skybur** [comment]
Well, one of my use case is like this

- I have Screen A with list of items loaded from network and cached.
- I click an item to navigate to Screen B which is the item detail screen.
- There I can update the item properties without popping the detail screen
- I want to refresh the Screen A's item list (some of the properties are displayed there) so I provided the callback to Screen B to call when the item changed

I can't think any other way besides providing callback.

**Comment by Milad-Akarie** [comment]
@skybur You could have a shared model/bloc between your master/details screens or if you're going for a simpler approach you could have your DetailsScreenRoute return the edited ITEM then update your list from there 
```dart
var editedItem = await Router.navigator.push(Router.detailsScreen);
// inside of your details screen pop with Results
  Router.navigator.pop([editedItem]);
```

**Comment by skybur** [comment]
I tried using `Navigator.pop` to pass back value, but I don't quite like it, because I think it's not descriptive enough about what will be returned. I mean I need to see inside the class that calls `Navigator.pop` to know what will be returned. On the other hand if I used callback to return the value, I know what will be returned. That's just me though and maybe that's not a good practice.

However I'm gonna try your suggestion since I'm totally gonna use this package for routing, thanks!

**Comment by Milad-Akarie** [comment]
@skybur, You can easily type your returned results

```dart
@MaterialRoute(returnType: ITEMTYPE)
DetailsScreen detialsScreen;
....

Router.navigator.pushNamed<ITEMTYPE>()
Router.navigator.pop<ITEMTYPE>(item)
```
I hope this helps. and thanks for reporting the bug, I'll let you know when it's fixed.

**Comment by skybur** [comment]
I'm wondering what that `returnType` is used for. So that's how I use it! 
I think that improves the descriptiveness when using `Navigator.pop` since I can just see in the router class what will be returned. Thanks again!

**Comment by Milad-Akarie** [comment]
@skybur You're welcome.

**Comment by Milad-Akarie** [comment]
@skybur  finally fixed in auto_route_generator 0.4.10 Thanks @nateshmbhat
